src/schema-generator/conditional-render.ts

```diff
@@ -98,19 +151,26 @@ const canRender = (
 	yupSchema: ObjectShape,
 	formValues: TypeOfShape<Assign<ObjectShape, ObjectShape>>
 ) => {
-	if (isEmpty(renderRules)) return true;
+	if (isEmpty(renderRules)) return { isValidWithoutShownRule: true, isValidWithShownRule: false, sourceIds: [] };
 
-	let isValid = false;
+	const sourceIds = [];
+	let isValidWithoutShownRule = false;
+	let isValidWithShownRule = false;
```

Comment:
had some confusions with the 2 flags, e.g. why is it not valid without shown rule, vice-versa.

instead of this can we keep the `isValid` flag and introduce `hasShownRules` and `hasNonShownRules`?

in `parseChildrenConditionalRenders()` we can update the conditions:
```
if (isValid && hasShownRules && !nonShownRules) {
	fieldsWithShownRule[id] = { childIds: listAllChildIds(componentSchema), sourceIds: sourceIds };
}

if (isValid) {
	switch (componentSchema.uiType) {
		case "checkbox":
		case "radio":
			(componentSchema as ICheckboxSchema | IRadioSchema).options.forEach((option) => {
				parseIfValidChildren(option.children);
			});
			break;
		default:
			parseIfValidChildren(componentSchema.children as Record<string, TComponentSchema>);
			break;
	}
} else {
	const hiddenFieldIdList = [id, ...listAllChildIds(componentSchema)];
	hiddenFieldIdList.forEach((hiddenFieldId) => {
		parsedYupSchema[hiddenFieldId] = Yup.mixed()
			.meta({ hidden: true })
			.test("empty", ERROR_MESSAGES.UNSPECIFIED_FIELD(hiddenFieldId), (value) => value === undefined);
	});
}
```